Shared.EventStore/EventStore/AggregateRepository.cs

```diff
@@ -15,37 +21,47 @@ public sealed class AggregateRepository<T> : IAggregateRepository<T> where T : A
         /// The context
         /// </summary>
         private readonly IEventStoreContext Context;
-        
+
         #endregion
 
         #region Constructors
-        
+
         /// <summary>
-        /// Initializes a new instance of the <see cref="AggregateRepository{T}"/> class.
+        /// Initializes a new instance of the <see cref="AggregateRepository{T}" /> class.
         /// </summary>
+        /// <param name="eventStoreContext">The event store context.</param>
         public AggregateRepository(IEventStoreContext eventStoreContext)
         {
             this.Context = eventStoreContext;
+            this.Context.TraceGenerated += this.Context_TraceGenerated;
         }
-        
+
         #endregion
 
-        #region Public Methods
+        #region Events
 
-        #region public async Task<T> GetLatestVersion(Guid aggregateId,CancellationToken cancellationToken)
         /// <summary>
-        /// get latest version as an asynchronous operation.
+        /// Occurs when trace is generated.
+        /// </summary>
+        public event TraceHandler TraceGenerated;
```

Comment:
Codacy found an issue: [Remove the unused event 'TraceGenerated' or invoke it.](https://app.codacy.com/manual/stuart_ferguson1/Shared/pullRequest?prid=5768660)